pkg/pillar/cmd/zedrouter/acl.go

```diff
@@ -254,162 +316,159 @@ func determineIpVer(isMgmt bool, bridgeIP string) int {
 	}
 }
 
-func applyACLRules(rules IptablesRuleList, bridgeName string, vifName string,
-	isMgmt bool, ipVer int, appIP string) error {
-
-	log.Debugf("applyACLRules: bridgeName %s ipVer %d appIP %s with %d rules\n",
-		bridgeName, ipVer, appIP, len(rules))
+func applyACLRules(aclArgs types.AppNetworkACLArgs,
+	rules types.IPTablesRuleList) (types.IPTablesRuleList, error) {
 	var err error
-	for _, rule := range rules {
-		log.Debugf("createACLConfiglet: rule %v\n", rule)
-		args := rulePrefix("-A", isMgmt, ipVer, vifName, appIP, rule)
-		if args == nil {
-			log.Debugf("createACLConfiglet: skipping rule %v\n",
-				rule)
+	var activeRules types.IPTablesRuleList
+	log.Debugf("applyACLRules: ipVer %d, bridgeName %s appIP %s with %d rules\n",
+		aclArgs.IPVer, aclArgs.BridgeName, aclArgs.AppIP, len(rules))
+	numRules := len(rules)
+	for numRules > 0 {
+		numRules--
+		rule := rules[numRules]
+		log.Debugf("createACLConfiglet: add rule %v\n", rule)
+		if err := rulePrefix(aclArgs, &rule); err != nil {
+			log.Debugf("createACLConfiglet: skipping rule %v\n", rule)
 			continue
 		}
-		args = append(args, rule...)
-		if ipVer == 4 {
-			err = iptables.IptableCmd(args...)
-		} else if ipVer == 6 {
-			err = iptables.Ip6tableCmd(args...)
+		err = executeIPTablesRule("-I", rule)
```

Comment:
1. This should be -A not -I so that the drop rule is always at the end.
2. rulePrefix is already adding the -A.

So I don't see why you need an operation argument to executeIPTablesRule

Reply:
rulePrefix, is not adding the operation flag any more. Currently, I have  instrumented the rule add with "-I"  happening in the reverse order,  to add the rules at the top of the chains. Not sure, whether new set of iptables rules, should be placed on top or bottom of the table. For placing them in the bottom, we can do -A.

Reply:
Does that work with the droprules, which is a pair of rules with the first being log and the second being drop? If you insert those in reverse order the log won't do anything, right?

Reply:
We are creating the set of rules by calling aclToRules followed by aclDropRules inside createACLConfiglet() function.  And we insert rules to the top of the chain in the reverse order, with each insertion, the last inserted is pushed down(kind of stack push operation). This way the drop rules are inserted first, followed by the aclrules. Hence the drop rules will be down in the rule stack.